catalyst_voices/packages/catalyst_voices_services/lib/src/crypto/crypto_service.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:flutter/foundation.dart';
+
+/// An abstract interface that defines cryptographic operations such as
+/// key derivation, encryption, decryption, and key verification.
+abstract interface class CryptoService {
```

Comment:
Seems similar to KeyDerivation class, at least the description. In the future we need to rename KeyDerivation to CardanoKeyDerivation because we will need to change the algorithm to use the same as cardano does.

Reply:
I'll add TODO about it. 

What we could do is exposing `KeyDerivation` interface and make `CryptoService` implement it with delegate to `Pbkdf2KeyDerivation`.

Ill make those changes in next PR.

Reply:
We'll wait with extraction of `KeyDerivation` interface until we'll understand more what kind of methods we'll need